### PR TITLE
tests and solution for challenge 5

### PR DESCRIPTION
**Pull request by jacobgarner**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -91,3 +91,27 @@ describe("/api/articles/:article_id", () => {
...
+            response.body.articles.forEach((article) => {
+            expect(article).toMatchObject({
+                author: expect.any(String),
+                title: expect.any(String),
+                article_id: expect.any(Number),
+                body: expect.any(String),
+                topic: expect.any(String),
+                created_at: expect.any(String),
+                votes: expect.any(Number),
+                article_img_url: expect.any(String),
```
Comment (nc-jonathan-mcguire) [inline __tests__/app.test.js:115]: Need a comment count 

comment_count, which is the total count of all the comments with this article_id. You should make use of queries to the database in order to achieve this.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -91,3 +91,27 @@ describe("/api/articles/:article_id", () => {
...
+              });
+        });
```
Comment (nc-jonathan-mcguire) [inline __tests__/app.test.js:118]: Need to test for this from the ticket as well

In addition:

the articles should be sorted by date in descending order.
there should not be a body property present on any of the article objects.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -91,3 +91,27 @@ describe("/api/articles/:article_id", () => {
...
+      return request(app)
+        .get("/api/articles")
+        .then((response) => {
+            expect(response.body.articles.length).toBe(13)
+            response.body.articles.forEach((article) => {
+            expect(article).toMatchObject({
+                author: expect.any(String),
+                title: expect.any(String),
+                article_id: expect.any(Number),
+                body: expect.any(String),
```
Comment (nc-jonathan-mcguire) [inline __tests__/app.test.js]: Shouldnt be a body key

**Review by nc-jonathan-mcguire — CHANGES_REQUESTED** [review]
Two bits missing from the ticket:

- [ ] comment count
- [ ] Shouldnt be body key
- [ ] the articles should be sorted by date in descending order. - make sure to test for this

File: models/topicsModel.js [hunk 4]
```diff
@@ -1,5 +1,7 @@
 const db = require("../db/connection");
 const fs = require("fs/promises");
+const articles = require("../db/data/test-data/articles");
```
Comment (ghost) [inline models/topicsModel.js:3]: What are these imports doing???

**Review by ghost — APPROVED** [review]
Not sure why you've imported test data into your models, you're not going to be using it. Make sure to remove that as they'll be unused variables and possibly cause your build to fail.

But good join in your sql, merge and move on.

**Pull request merged**